controllers/nutanixmachine_controller.go

```diff
@@ -1118,40 +1159,15 @@ func (r *NutanixMachineReconciler) GetSubnetAndPEUUIDs(rctx *nctx.MachineContext
 	if rctx == nil {
 		return "", nil, fmt.Errorf("cannot create machine config if machine context is nil")
 	}
-	log := ctrl.LoggerFrom(rctx.Context)
-	if rctx.Machine.Spec.FailureDomain == nil || *rctx.Machine.Spec.FailureDomain == "" {
-		log.V(1).Info("no failure domain found on machine. Directly searching for Prism Element cluster")
-		if rctx.NutanixMachine.Spec.Cluster.Name == nil && rctx.NutanixMachine.Spec.Cluster.UUID == nil {
-			return "", nil, fmt.Errorf("cluster name or uuid must be passed if failure domain is not configured")
-		}
-		if len(rctx.NutanixMachine.Spec.Subnets) == 0 {
-			return "", nil, fmt.Errorf("subnets must be passed if failure domain is not configured")
-		}
-		peUUID, err := GetPEUUID(rctx.Context, rctx.NutanixClient, rctx.NutanixMachine.Spec.Cluster.Name, rctx.NutanixMachine.Spec.Cluster.UUID)
-		if err != nil {
-			return "", nil, err
-		}
-		subnetUUIDs, err := GetSubnetUUIDList(rctx.Context, rctx.NutanixClient, rctx.NutanixMachine.Spec.Subnets, peUUID)
-		if err != nil {
-			return "", nil, err
-		}
-		return peUUID, subnetUUIDs, nil
-	}
 
-	log.V(1).Info("failure domain config found. Ignoring cluster config on machine object (if any)")
-
-	failureDomainName := *rctx.Machine.Spec.FailureDomain
-	failureDomain, err := GetFailureDomain(failureDomainName, rctx.NutanixCluster)
-	if err != nil {
-		return "", nil, fmt.Errorf("failed to find failure domain %s", failureDomainName)
-	}
-	peUUID, err := GetPEUUID(rctx.Context, rctx.NutanixClient, failureDomain.Cluster.Name, failureDomain.Cluster.UUID)
+	peUUID, err := GetPEUUID(rctx.Context, rctx.NutanixClient, rctx.NutanixMachine.Spec.Cluster.Name, rctx.NutanixMachine.Spec.Cluster.UUID)
 	if err != nil {
-		return "", nil, fmt.Errorf("failed to find prism element uuid for failure domain %s", failureDomainName)
+		return "", nil, err
 	}
-	subnetUUIDs, err := GetSubnetUUIDList(rctx.Context, rctx.NutanixClient, failureDomain.Subnets, peUUID)
+
+	subnetUUIDs, err := GetSubnetUUIDList(rctx.Context, rctx.NutanixClient, rctx.NutanixMachine.Spec.Subnets, peUUID)
 	if err != nil {
-		return "", nil, fmt.Errorf("failed to find subnet uuids for failure domain %s", failureDomainName)
+		return "", nil, err
```

Comment:
Any reason we needed to remove wrapping the errors?

Reply:
Here in this function, failureDomain is not involved any more. So we don't need to wrap the error message to include the failureDomainName.